tvplayer/src/main/java/com/kaltura/tvplayer/KalturaPlayer.java

```diff
@@ -1308,6 +1332,74 @@ private void sendKavaPlayRequest(String entryId) {
         NetworkUtils.sendKavaAnalytics(context, partnerId, entryId, NetworkUtils.KAVA_EVENT_PLAY_REQUEST, sessionId);
     }
 
+    private PKAdvertisingController getPkAdvertisingController() {
+        if (pkAdvertisingController == null) {
```

Comment:
should we create one if it is not needed

Reply:
Yes, Done